### PR TITLE
Remove persistentVolumeClaim by volumeClaimTemplate

### LINKED ISSUE #181: Remove PVC definition alone. Generate it with pipelinerun together
/kind feature

**Description:**
[A clear and concise description of what your proposal. What problem does it solve?]
Create pvc for pipelinerun if using big data passing. As we used workspaces in tekton pipelines which depends on it. 

TODO:
Remove PVC if Tekton version > = 0.12, use 'volumeClaimTemplate' instead

 'volumeClaimTemplate'  is not support if Tekton version < 0.12

Dependency:
 - [x] PR #143
 - [x] PR #172

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

/assign

### PR DESCRIPTION
**Pull request by fenglixa**
**Which issue is resolved by this Pull Request:** 
Resolves #181

**Description of your changes:**
Remove persistentVolumeClaim by volumeClaimTemplate, since we already used the Tekton version > 0.12
**Environment tested:**
```
(.venv)  fengli@fenglis-MacBook-Pro  ~/go/src/github.com/kubeflow/kfp-tekton/sdk/python/tests   remove_pvc  kc get po |grep file
file-passing-pipelines-gen-params-mv8h8-pod-hqhnv         0/1     Completed   0          2d13h
file-passing-pipelines-print-params-25vkf-pod-zvpsm       0/1     Completed   0          2d13h
file-passing-pipelines-print-text-2-qsccx-pod-s5fkx       0/1     Completed   0          2d13h
file-passing-pipelines-print-text-3-xj69j-pod-mxvl8       0/1     Completed   0          2d13h
file-passing-pipelines-print-text-4-kfnr7-pod-9rfxq       0/1     Completed   0          2d13h
file-passing-pipelines-print-text-5-l9snc-pod-jtbhc       0/1     Completed   0          2d13h
file-passing-pipelines-print-text-dclrj-pod-w9lfj         0/1     Completed   0          2d13h
file-passing-pipelines-repeat-line-x45z5-pod-44lbv        0/1     Completed   0          2d13h
file-passing-pipelines-split-text-lines-crb4d-pod-hjdw7   0/2     Completed   0          2d13h
file-passing-pipelines-sum-numbers-fpdvm-pod-kq6r7        0/1     Completed   0          2d13h
file-passing-pipelines-write-numbers-6w9rb-pod-f8tx5      0/1     Completed   0          2d13h
(.venv)  fengli@fenglis-MacBook-Pro  ~/go/src/github.com/kubeflow/kfp-tekton/sdk/python/tests   remove_pvc  tkn pr describe file-passing-pipelines
Name:        file-passing-pipelines
Namespace:   feng-test
Timeout:     1h0m0s
Labels:
 tekton.dev/pipeline=file-passing-pipelines

🌡️  Status

STARTED      DURATION    STATUS
2 days ago   2 minutes   Succeeded

📦 Resources

 No resources

⚓ Params

 No params

🗂  Taskruns

 NAME                                              TASK NAME          STARTED      DURATION     STATUS
 ∙ file-passing-pipelines-print-text-5-l9snc       print-text-5       2 days ago   6 seconds    Succeeded
 ∙ file-passing-pipelines-print-text-3-xj69j       print-text-3       2 days ago   10 seconds   Succeeded
 ∙ file-passing-pipelines-print-text-2-qsccx       print-text-2       2 days ago   1 minute     Succeeded
 ∙ file-passing-pipelines-print-text-4-kfnr7       print-text-4       2 days ago   13 seconds   Succeeded
 ∙ file-passing-pipelines-sum-numbers-fpdvm        sum-numbers        2 days ago   15 seconds   Succeeded
 ∙ file-passing-pipelines-print-text-dclrj         print-text         2 days ago   1 minute     Succeeded
 ∙ file-passing-pipelines-print-params-25vkf       print-params       2 days ago   1 minute     Succeeded
 ∙ file-passing-pipelines-write-numbers-6w9rb      write-numbers      2 days ago   15 seconds   Succeeded
 ∙ file-passing-pipelines-gen-params-mv8h8         gen-params         2 days ago   11 seconds   Succeeded
 ∙ file-passing-pipelines-split-text-lines-crb4d   split-text-lines   2 days ago   22 seconds   Succeeded
 ∙ file-passing-pipelines-repeat-line-x45z5        repeat-line        2 days ago   14 seconds   Succeeded
```

### CODE REVIEW

File: sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py [hunk 1]
```diff
@@ -453,10 +443,19 @@ def big_data_passing_pipelinerun(name: str, pr: dict, pw: set):
...
         pr.get('spec', {}).setdefault('workspaces', [])
+        # Change persistentVolumeClaim to volumeClaimTemplate which need Tekton version > 0.12
+        # User could modify default size of storage in the yaml file mannually if necessary.
         pr['spec']['workspaces'].append({
             "name": pipelinerun_name,
-            "persistentVolumeClaim": {
-                "claimName": pipelinerun_name
+            "volumeClaimTemplate": {
+                "spec": {
+                    "accessModes": ["ReadWriteMany"],
```
Comment (Tomcli) [inline sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py:452]: Many of the IBM Cloud users are using block storage, can you make the default to `ReadWriteOnce`? Thanks.

Comment (fenglixa) [reply sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py:452]: Hi @Tomcli , if `ReadWriteOnce` was set, then some cases may not work for big data pipelineruns. e.g. different tasks in the pipeline read/write the data to storage in the same time.

Comment (Tomcli) [reply sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py:452]: Hi @fenglixa, then can we have some environment variable to change the accessModes? thanks

Comment (fenglixa) [reply sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py:452]: OK, @Tomcli , I addressed it as your suggestion, make as an environment variable. make as resolve conversatioon now

**Comment by fenglixa** [comment]
@Tomcli , Thanks for your comments, I addressed it just now. is it OK for you now? Thanks!

**Comment by Tomcli** [comment]
Thanks @fenglixa, we need to document this along with updating the samples you have. But those can be done in a separate PR.
/lgtm

**Pull request merged**